src/xtgeo/interfaces/rms/_rms_base.py

```diff
@@ -0,0 +1,86 @@
+"""Shared base for RMS RegularSurface reader/writer."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from xtgeo.common.log import null_logger
+from xtgeo.interfaces.rms.rmsapi_utils import (
+    RmsApiUtils,
+    _StorageTypeRegularSurface as StorageType,
+)
+
+if TYPE_CHECKING:
+    from xtgeo.interfaces.rms._rmsapi_package import RmsProjectType
+
+logger = null_logger(__name__)
+
+__all__ = ["_BaseRmsDataRW"]
+
+
+class _BaseRmsDataRW:
+    """
+    Common init, input validation etc for read/write RMS RegularSurface, Points, etc.
+    """
+
+    def __init__(
+        self,
+        project: RmsProjectType,
+        name: str,
+        category: str,
+        stype: str,
+        realisation: int = 0,
+    ):
+        self.project = project
+        self.name = name
+        self.category = category
+        self.stype = stype
+        self.realisation = realisation
+
+        # Internal state
+        self._rmsapi_utils: RmsApiUtils | None = None
+        self._stype_enum: StorageType | None = None
+
+    def _init_utils(self, readonly: bool) -> None:
+        """Create RmsApiUtils with the selected readonly mode."""
+        self._rmsapi_utils = RmsApiUtils(self.project, readonly=readonly)
+
+    def _validate_inputs(self) -> StorageType:
+        """Validate input parameters and return StorageType enum."""
+        stype_lower = self.stype.lower()
+
+        if stype_lower not in StorageType.values():
+            raise ValueError(
+                f"Given stype '{stype_lower}' is not supported. "
+                f"Legal stypes are: {StorageType.values()}"
+            )
+
+        stype_enum = StorageType(stype_lower)
+
+        if not self.name:
+            raise ValueError("The name is missing or empty.")
+
+        if (
+            stype_enum in (StorageType.HORIZONS, StorageType.ZONES)
+            and not self.category
+        ):
+            raise ValueError(
+                "Need to specify both name and category for horizons and zones"
+            )
+
+        if stype_enum == StorageType.GENERAL2D_DATA:
```

Comment:
might be that this can be removed since we support only python 3.11+ which support this in RMSAPI